pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test: TestLogic_pg_builtins failed

pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test.TestLogic_pg_builtins [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_SqlLogicTestHighVModuleNightlyBazel/13407856?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_SqlLogicTestHighVModuleNightlyBazel/13407856?buildTab=artifacts#/) on master @ [ba84e57f38d31b85758b6c9ff305bf7d07752f25](https://github.com/cockroachdb/cockroach/commits/ba84e57f38d31b85758b6c9ff305bf7d07752f25):


```
[08:50:26] COMMIT;
[08:50:26] SELECT current_setting('vacuum_cost_delay', false);
[08:50:26] 	-- XFAIL;
[08:50:26] SHOW application_name;
[08:50:26] 	-- OK;
[08:50:26] SELECT set_config('application_name', 'woo', false);
[08:50:26] 	-- OK;
[08:50:26] SHOW application_name;
[08:50:26] 	-- OK;
[08:50:26] SELECT
  set_config('application_name', 'foo', false),
  set_config('statement_timeout', '60s', false),
  set_config(NULL, 'foo', false);
[08:50:26] 	-- OK;
[08:50:26] SELECT set_config('application_name', 'woo', true);
[08:50:26] 	-- OK;
[08:50:26] SELECT current_setting('application_name');
[08:50:26] 	-- OK;
[08:50:26] BEGIN;
[08:50:26] SELECT set_config('application_name', 'woo', true);
[08:50:26] 	-- OK;
[08:50:26] SELECT current_setting('application_name'), current_setting('statement_timeout');
[08:50:26] 	-- OK;
[08:50:26] COMMIT;
[08:50:26] SELECT current_setting('application_name');
[08:50:26] 	-- OK;
[08:50:26] SELECT pg_catalog.set_config('woo', 'woo', false);
[08:50:26] 	-- XFAIL;
[08:50:26] SELECT set_config('vacuum_cost_delay', '0', false);
[08:50:26] 	-- XFAIL;
[08:50:26] SELECT pg_catalog.set_config('search_path', '', false);
[08:50:26] 	-- OK;
[08:50:26] SHOW search_path;
[08:50:26] 	-- FAIL
    logic.go:2927: 
         
        /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/4261/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test_/fakedist-disk_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/pg_builtins:410: SHOW search_path
        expected:
            ·
            
        but found (query options: "") :
            "$user", public
            
[08:50:26] 	-- FAIL
    logic.go:2184: 
         /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/4261/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test_/fakedist-disk_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/pg_builtins:415: too many errors encountered, skipping the rest of the input
[08:50:26] --- done: /home/roach/.cache/bazel/_bazel_roach/c5a4e7d36696d9cd970af2045211a7df/sandbox/processwrapper-sandbox/4261/execroot/com_github_cockroachdb_cockroach/bazel-out/k8-fastbuild/bin/pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test_/fakedist-disk_test.runfiles/com_github_cockroachdb_cockroach/pkg/sql/logictest/testdata/logic_test/pg_builtins with config fakedist-disk: 63 tests, 2 failures
[08:50:29] --- total progress: 63 statements
--- total: 63 tests, 2 failures
    --- FAIL: TestLogic_pg_builtins/end (1.46s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
/cc @cockroachdb/sql-queries
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogic_pg_builtins.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>


Jira issue: CRDB-35139

pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test.TestLogic_pg_builtins [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_SqlLogicTestHighVModuleNightlyBazel/13422228?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_SqlLogicTestHighVModuleNightlyBazel/13422228?buildTab=artifacts#/) on master @ [0039f06034104aeafa19d4f2e1e020e433663e1b](https://github.com/cockroachdb/cockroach/commits/0039f06034104aeafa19d4f2e1e020e433663e1b):


```
large                              20       0               0       0               0       0               9       0               0         0               0           0
extent_avail                       28       0               0       0               0       0              13       0               0         0               0           0
extents_dirty                      26       0               0       0               0       0              13       0               0         0               0           0
extents_muzzy                      20       0               0       0               0       0               9       0               0         0               0           0
extents_retained                   28       0               0       0               0       0              13       0               0         0               0           0
decay_dirty                        24       0               0       0               0       0              12       0               0         0               0           0
decay_muzzy                        20       0               0       0               0       0               9       0               0         0               0           0
base                               45       0               0       0               0       0              11       0               0         0               0           0
tcache_list                        20       0               0       0               0       0               9       0               0         0               0           0
hpa_shard                           0       0               0       0               0       0               0       0               0         0               0           0
hpa_shard_grow                      0       0               0       0               0       0               0       0               0         0               0           0
hpa_sec                             0       0               0       0               0       0               0       0               0         0               0           0
bins:           size ind    allocated      nmalloc (#/sec)      ndalloc (#/sec)    nrequests   (#/sec)  nshards      curregs     curslabs  nonfull_slabs regs pgs   util       nfills (#/sec)     nflushes (#/sec)       nslabs     nreslabs (#/sec)      n_lock_ops (#/sec)       n_waiting (#/sec)      n_spin_acq (#/sec)  n_owner_switch (#/sec)   total_wait_ns   (#/sec)     max_wait_ns  max_n_thds
                     ---
large:          size ind    allocated      nmalloc (#/sec)      ndalloc (#/sec)    nrequests (#/sec)  curlextents
                     ---
             8388608  72            0            2       0            2       0            2       0            0
                     ---
            16777216  76            0            1       0            1       0            1       0            0
                     ---
extents:        size ind       ndirty        dirty       nmuzzy        muzzy    nretained     retained       ntotal        total
                     ---
            10485760  40            0            0            0            0            1     10485760            1     10485760
                     ---
            20971520  44            0            0            0            0            1     20971520            1     20971520
                     ---
Bytes in small extent cache: 0
HPA shard stats:
  Purge passes: 0 (0 / sec)
  Purges: 0 (0 / sec)
  Hugeifies: 0 (0 / sec)
  Dehugifies: 0 (0 / sec)

  In full slabs:
      npageslabs: 0 huge, 0 nonhuge
      nactive: 0 huge, 0 nonhuge 
      ndirty: 0 huge, 0 nonhuge 
      nretained: 0 huge, 0 nonhuge 
  In empty slabs:
      npageslabs: 0 huge, 0 nonhuge
      nactive: 0 huge, 0 nonhuge 
      ndirty: 0 huge, 0 nonhuge 
      nretained: 0 huge, 0 nonhuge 

                size ind npageslabs_huge    nactive_huge     ndirty_huge  npageslabs_nonhuge     nactive_nonhuge      ndirty_nonhuge   nretained_nonhuge
                     ---
--- End jemalloc statistics ---
[08:29:01] --- total progress: 63 statements
--- total: 63 tests, 2 failures
    --- FAIL: TestLogic_pg_builtins/end (1.67s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogic_pg_builtins.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

pkg/sql/logictest/tests/fakedist-disk/fakedist-disk_test.TestLogic_pg_builtins [failed](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_SqlLogicTestHighVModuleNightlyBazel/13433030?buildTab=log) with [artifacts](https://teamcity.cockroachdb.com/buildConfiguration/Cockroach_Nightlies_SqlLogicTestHighVModuleNightlyBazel/13433030?buildTab=artifacts#/) on master @ [677cd7db760cc1a0793064da7c2e6c03b14e33a6](https://github.com/cockroachdb/cockroach/commits/677cd7db760cc1a0793064da7c2e6c03b14e33a6):


```
large                              20       0               0       0               0       0              10       0               0         0               0           0
extent_avail                       28       0               0       0               0       0              14       0               0         0               0           0
extents_dirty                      26       0               0       0               0       0              14       0               0         0               0           0
extents_muzzy                      20       0               0       0               0       0              10       0               0         0               0           0
extents_retained                   28       0               0       0               0       0              14       0               0         0               0           0
decay_dirty                        24       0               0       0               0       0              13       0               0         0               0           0
decay_muzzy                        20       0               0       0               0       0              10       0               0         0               0           0
base                               45       0               0       0               0       0              12       0               0         0               0           0
tcache_list                        20       0               0       0               0       0              10       0               0         0               0           0
hpa_shard                           0       0               0       0               0       0               0       0               0         0               0           0
hpa_shard_grow                      0       0               0       0               0       0               0       0               0         0               0           0
hpa_sec                             0       0               0       0               0       0               0       0               0         0               0           0
bins:           size ind    allocated      nmalloc (#/sec)      ndalloc (#/sec)    nrequests   (#/sec)  nshards      curregs     curslabs  nonfull_slabs regs pgs   util       nfills (#/sec)     nflushes (#/sec)       nslabs     nreslabs (#/sec)      n_lock_ops (#/sec)       n_waiting (#/sec)      n_spin_acq (#/sec)  n_owner_switch (#/sec)   total_wait_ns   (#/sec)     max_wait_ns  max_n_thds
                     ---
large:          size ind    allocated      nmalloc (#/sec)      ndalloc (#/sec)    nrequests (#/sec)  curlextents
                     ---
             8388608  72            0            2       0            2       0            2       0            0
                     ---
            16777216  76            0            1       0            1       0            1       0            0
                     ---
extents:        size ind       ndirty        dirty       nmuzzy        muzzy    nretained     retained       ntotal        total
                     ---
            10485760  40            0            0            0            0            1     10485760            1     10485760
                     ---
            20971520  44            0            0            0            0            1     20971520            1     20971520
                     ---
Bytes in small extent cache: 0
HPA shard stats:
  Purge passes: 0 (0 / sec)
  Purges: 0 (0 / sec)
  Hugeifies: 0 (0 / sec)
  Dehugifies: 0 (0 / sec)

  In full slabs:
      npageslabs: 0 huge, 0 nonhuge
      nactive: 0 huge, 0 nonhuge 
      ndirty: 0 huge, 0 nonhuge 
      nretained: 0 huge, 0 nonhuge 
  In empty slabs:
      npageslabs: 0 huge, 0 nonhuge
      nactive: 0 huge, 0 nonhuge 
      ndirty: 0 huge, 0 nonhuge 
      nretained: 0 huge, 0 nonhuge 

                size ind npageslabs_huge    nactive_huge     ndirty_huge  npageslabs_nonhuge     nactive_nonhuge      ndirty_nonhuge   nretained_nonhuge
                     ---
--- End jemalloc statistics ---
[08:47:36] --- total progress: 63 statements
--- total: 63 tests, 2 failures
    --- FAIL: TestLogic_pg_builtins/end (1.69s)
```
<details><summary>Help</summary>
<p>

See also: [How To Investigate a Go Test Failure \(internal\)](https://cockroachlabs.atlassian.net/l/c/HgfXfJgM)
</p>
</details>
<sub>

[This test on roachdash](https://roachdash.crdb.dev/?filter=status:open%20t:.*TestLogic_pg_builtins.*&sort=title+created&display=lastcommented+project) | [Improve this report!](https://github.com/cockroachdb/cockroach/tree/master/pkg/cmd/internal/issues)
</sub>

--- a/pkg/sql/conn_executor.go
+++ b/pkg/sql/conn_executor.go
@@ -980,6 +980,7 @@ func newSessionData(args SessionArgs) *sessiondata.SessionData {
 			sd.CustomOptions[k] = v
 		}
 	}
+	sd.SearchPath = sessiondata.DefaultSearchPathForUser(sd.User())
 	populateMinimalSessionData(sd)
 	return sd
 }
@@ -992,9 +993,6 @@ func populateMinimalSessionData(sd *sessiondata.SessionData) {
 	}
 	if sd.Location == nil {
 		sd.Location = time.UTC
-	}
-	if len(sd.SearchPath.GetPathArray()) == 0 {
-		sd.SearchPath = sessiondata.DefaultSearchPathForUser(sd.User())
 	}
 }
 

--- a/pkg/sql/internal.go
+++ b/pkg/sql/internal.go
@@ -1668,6 +1668,7 @@ func (ief *InternalDB) newInternalExecutorWithTxn(
 	if sd == nil {
 		sd = NewInternalSessionData(ctx, settings, "" /* opName */)
 		sd.UserProto = username.RootUserName().EncodeProto()
+		sd.SearchPath = sessiondata.DefaultSearchPathForUser(sd.User())
 	}
 
 	schemaChangerState := &SchemaChangerState{

